cvpy/tests/test_cvat_project.py

```diff
@@ -72,6 +78,96 @@ def test_cvat_project_add_task(self):
+        cvat_project = CVATProject(url=url, cas_connection=cas_connection, credentials=credentials,
+                                   project_name=project_name, annotation_type=annotation_type,
+                                   labels=labels)
+
+        # Post the images to the CVATProject.
+        cvat_project.post_images(image_table_encoded)
+        cvat_project.post_images(image_table_decoded)
+
+        # Note: CVAT will throw an internal server error exception if we attempt to access the tasks
+        #       data to soon after an upload. So we have a sleep here to prevent the issue.
+        time.sleep(5)
+
+        # Get the CVATTask objects we created and verify what was populated.
+        tasks = cvat_project.get_tasks()
+        self.assertEqual(len(tasks), 2)
+
+        for task in tasks:
+            # Get the task metadata from CVAT and then verify that it matches the tasks post_images created.
+            self.assertIsNotNone(task.task_id)
+
+            response = requests.get(f'{url}/api/tasks/{task.task_id}/data/meta', headers=credentials.get_auth_header())
+            self.assertEqual(response.status_code, HTTPStatus.OK)
+            self.assertEqual(response.json()['size'], task.image_table.table.tableinfo().TableInfo.Rows.values[0])
+            self.assertEqual(response.json()['start_frame'], task.start_image_id)
+            self.assertEqual(response.json()['stop_frame'], task.end_image_id)
+            
+            # Verify the image data and metadata between CAS and CVAT.
+            cvat_frames = response.json()['frames']
+            cas_images = task.image_table.table.fetchImages(fetchImagesVars=["_id_", "_type_"]).Images
+            for index, frame_number in enumerate(range(task.start_image_id, task.end_image_id+1)):
+                
+                # Check the image names.
+                expected_extension = 'jpg' if (task.image_table.has_decoded_images()) else cas_images.iloc[index]._type_
+                expected_name = f'{cas_images.iloc[index]._id_}.{expected_extension}'
+                self.assertEqual(cvat_frames[index]['name'], expected_name)
+
+                # Check the image bytes. (We have to GET the image from CVAT first.) 
+                response = requests.get(f'{url}/api/tasks/{task.task_id}/data', 
+                                        headers=credentials.get_auth_header(),
+                                        params=dict(quality='original', number=frame_number, type='frame'))
+            
+                self.assertEqual(response.status_code, HTTPStatus.OK)
+                cvat_image_bytes = io.BytesIO(response.content)
+
+                cas_image_bytes = io.BytesIO()
+                pillow_format = 'JPEG' if expected_extension == 'jpg' else expected_extension
+                cas_images.iloc[index].Image.save(cas_image_bytes, format=pillow_format)
+                self.assertEqual(cvat_image_bytes.getvalue(), cas_image_bytes.getvalue())
```

Comment:
This looks great! 

I will suggest one adding one more check. The above logic is getting tasks from CVPy cvat_project object and verifying those tasks are correctly created in CVAT. However, I don't see a check for correct association of tasks with the project in CVAT. It will be good to add that check.

The tasks API returns project_id associated with the task. 
/api/tasks/{id}

Reply:
Good suggestion! I've decided to add that check to test_cvat_task.py as it's more related to to CVATTask._create_cvat_task() than CVATProject.post_images().